pkg/domain/infra/abi/system.go

```diff
@@ -83,11 +83,15 @@ func (ic *ContainerEngine) SetupRootless(_ context.Context, noMoveProcess bool)
 	}
 
 	// do it only after podman has already re-execed and running with uid==0.
-	hasCapSysAdmin, err := unshare.HasCapSysAdmin()
-	if err != nil {
-		return err
+	needsUserns := os.Geteuid() != 0
```

Comment:
above here there is 
```
if !rootless.IsRootless() {
	return nil
}
```

so this check here seems unnecessary

Reply:
thanks.

Pushed a new version.